minari/minari_dataset.py

```
@@ -1,141 +1,80 @@
         """
-        self._data_path = data_path
-        self._extra_data_id = 0
-        with h5py.File(self._data_path, "r") as f:
-            self._flatten_observations = f.attrs["flatten_observation"]
-            self._flatten_actions = f.attrs["flatten_action"]
-            self._env_spec = EnvSpec.from_json(f.attrs["env_spec"])
-
-            self._total_episodes: int = f.attrs["total_episodes"]
-            self._total_steps: int = f.attrs["total_steps"]
-
-            self._dataset_name = f.attrs["dataset_name"]
-            self._combined_datasets = f.attrs.get("combined_datasets")
-
-            env = gym.make(self._env_spec)
-
-            self._observation_space = env.observation_space
-            self._action_space = env.action_space
-
-            env.close()
-
-        self._episode_idx = list(range(self._total_episodes))
-
-    def recover_environment(self):
-        """Recover the Gymnasium environment used to create the dataset.
-
-        Returns:
-            environment: Gymnasium environment
-        """
-        return gym.make(self._env_spec)
-
-    @property
-    def flatten_observations(self) -> bool:
-        """If the observations have been flatten when creating the dataset."""
-        return self._flatten_observations
-
-    @property
-    def flatten_actions(self) -> bool:
-        """If the actions have been flatten when creating the dataset."""
-        return self._flatten_actions
-
-    @property
-    def observation_space(self):
-        """Original observation space of the environment before flatteining (if this is the case)."""
-        return self._observation_space
-
-    @property
-    def action_space(self):
-        """Original action space of the environment before flatteining (if this is the case)."""
-        return self._action_space
+        if isinstance(data, MinariStorage):
+            self._data = data
+        else:
+            self._data = MinariStorage(data)
```

Can we also raise an error if `data` is non of the allowed types?